File: Solutions/misc/zigzagConversion.py
def convert(s: str, numRows: int) -> str:
    if numRows == 1: return s

    rows = [""] * numRows
    currentRow = 0
    directionUp = True
    for char in s:
        rows[currentRow] += char
        if currentRow == numRows - 1 or currentRow == 0:
            directionUp = not directionUp
        
        if directionUp:
            currentRow -= 1
        else:
            currentRow += 1

    result = ""
    for row in rows:
        result += row
        
    return result

File: ZigzagConversion.ts
function convert(s: string, numRows: number): string {
    if (numRows == 1) return s;
    let rows: string[][] = new Array();
    for (let i=0; i < numRows; i++) {
        rows.push([]);
    }
    let currentRow = 0;
    let dirDown: boolean = true;

    for (let i=0; i < s.length; i++) {
        let c: string = s[i];
        rows[currentRow].push(c);
        currentRow += (dirDown ? 1 : -1)
        if (currentRow == numRows - 1) dirDown = false
        if (currentRow == 0) dirDown = true;
    }

    return rows.map((row) => row.join("")).join("");
};